src/catalog/src/builtin.rs

```diff
@@ -2862,15 +2862,51 @@ LEFT JOIN mz_catalog.mz_databases d ON d.id = s.database_id
 WHERE s.database_id IS NULL OR d.name = current_database()",
 };
 
+// Note: Databases, Roles, Clusters, Cluster Replicas, Secrets, and Connections are excluded from
+// this view.
 pub const PG_DESCRIPTION: BuiltinView = BuiltinView {
     name: "pg_description",
     schema: PG_CATALOG_SCHEMA,
-    sql: "CREATE VIEW pg_catalog.pg_description AS SELECT
-    c.oid as objoid,
-    NULL::pg_catalog.oid as classoid,
-    0::pg_catalog.int4 as objsubid,
-    NULL::pg_catalog.text as description
-FROM pg_catalog.pg_class c",
+    sql: "CREATE VIEW pg_catalog.pg_description AS
+SELECT
+	pg_obj.objoid,
+	pg_obj.classoid,
+	COALESCE(cmt.object_sub_id::INT4, 0) AS objsubid,
+	cmt.comment AS description
+FROM
+	(
+		SELECT
+			oid, (SELECT oid FROM mz_catalog.mz_objects WHERE name = 'pg_class')
+		FROM
+			pg_catalog.pg_class
+		UNION ALL
```

Comment:
> These unions make this a view that is difficult to reason about because they mix pg_ and mz_ catalogs.

Would changing `(SELECT oid FROM mz_catalog.mz_objects WHERE name = 'pg_class')` to `(SELECT oid FROM pg_catalog.pg_class WHERE relname = 'pg_class')` make it clearer? That's what I originally had but changed it as a perf optimization.

> Is it possible to rely only on mz_objects for this view?

@ParkMyCar had a similar question. If we want to remove schemas from this view then we can only rely on `mz_objects`. However, this view is meant to contain schemas as well, which means we need to include `mz_schemas`. I'm not opposed to omitting schemas if we want but then this view would not return the same result as PostgreSQL.

In terms of removing the references to `pg_catalog` views, we can replace

```SQL
		SELECT
			oid, (SELECT oid FROM mz_catalog.mz_objects WHERE name = 'pg_class')
		FROM
			pg_catalog.pg_class
		UNION ALL
			SELECT
				oid,
				(SELECT oid FROM mz_catalog.mz_objects WHERE name = 'pg_type')
			FROM
				pg_catalog.pg_type
		UNION ALL
			SELECT
				oid,
				(
					SELECT
						oid
					FROM
						mz_catalog.mz_objects
					WHERE
						name = 'pg_namespace'
				)
			FROM
				pg_catalog.pg_namespace
```

with something like
```SQL
		SELECT
			oid, (SELECT oid FROM mz_catalog.mz_objects WHERE name = 'pg_class')
		FROM
			mz_catalog.mz_objects
                WHERE
                        type IN ('table', 'view', 'index', 'materialize view', 'source')
		UNION ALL
			SELECT
				oid,
				(SELECT oid FROM mz_catalog.mz_objects WHERE name = 'pg_type')
			FROM
				mz_catalog.mz_objects
                        WHERE type = 'type'
		UNION ALL
			SELECT
				oid,
				(
					SELECT
						oid
					FROM
						mz_catalog.mz_objects
					WHERE
						name = 'pg_namespace'
				)
			FROM
				mz_catalog.mz_schemas
```
but in my opinion it becomes much less obvious that `objsubid` is correct.

> The pg docs for the view say the oid corresponds to pg_class.oid, but the unions here would mean that the oids might not appear in pg_class, right?

That depends on which OID you're referring to.

> objoid oid (references any OID column)
> The OID of the object this description pertains to

This is the OID of an object (in the PostgreSQL sense of object) and can appear in any catalog table, not just `pg_class`. In my local PG instance this OID can be found in one of the following tables:

```
postgres=# SELECT relname FROM (SELECT DISTINCT(classoid) FROM pg_description) o JOIN pg_class c ON o.classoid = c.oid;
    relname     
----------------
 pg_type
 pg_class
 pg_proc
 pg_operator
 pg_am
 pg_language
 pg_namespace
 pg_conversion
 pg_ts_config
 pg_ts_dict
 pg_ts_parser
 pg_ts_template
 pg_extension
 pg_collation
(14 rows)

```

> classoid oid (references [pg_class](https://www.postgresql.org/docs/16/catalog-pg-class.html).oid)
> The OID of the system catalog this object appears in

This is the OID of the catalog table that contains `objoid`. This OID will always appear in `pg_class`.

Reply:
This is all convincing enough to leave as is. Thanks for the explainer.

Reply:
Either way though, now that we're not rushing for the release, I'm going to hand this off to @ParkMyCar who maybe can think of a better view definition.

Maybe something like the following is more obvious:

```SQL
SELECT 
    oid, 
    id, 
    name, 
    CASE
        WHEN type IN ('table', 'view', ...) THEN (SELECT oid FROM mz_objects WHERE name = 'pg_class')
        WHEN type = 'type' THEN (SELECT oid FROM mz_objects WHERE name = 'pg_type')
        ELSE ???
    END,
    mz_comments.*
FROM mz_objects o
JOIN mz_comments c ON o.id = c.id AND o.type = c.type
UNION ALL
SELECT 
    oid, 
    id, 
    name, 
    (SELECT oid FROM mz_objects WHERE name = 'pg_namespace'),
    mz_comments.* 
FROM mz_schemas s
JOIN mz_comments c ON s.id = c.id AND s.type = c.type
```

Reply:
I rewrote the view definition using CTE's which I think makes it a bit easier to grok? Let me know what y'all think!